sdk/appconfiguration/app-configuration/src/models.ts

```diff
@@ -83,38 +82,16 @@ export interface HttpResponseFields {
   /**
    * The HTTP status code for the response
    */
-  statusCode: number;
+  statusCode: number | undefined;
 }
-
-/**
- * Parameters for adding a new configuration setting
- */
-export type AddConfigurationSettingParam<
-  T extends string | FeatureFlagValue | SecretReferenceValue = string
-> = ConfigurationSettingParam<T>;
-
-/**
- * Parameters for creating or updating a new configuration setting
- */
-export type SetConfigurationSettingParam<
-  T extends string | FeatureFlagValue | SecretReferenceValue = string
-> = ConfigurationSettingParam<T>;
-
-/**
- * Standard base response for getting, deleting or updating a configuration setting
- */
-export type ConfigurationSettingResponse<HeadersT> = ConfigurationSetting &
-  HttpResponseField<HeadersT> &
-  Pick<HeadersT, Exclude<keyof HeadersT, "eTag">>;
-
 /**
  * HTTP response related information - headers and raw body.
  */
 export interface HttpResponseField<HeadersT> {
   /**
    * The underlying HTTP response.
    */
-  _response: HttpResponse & {
+  _response: FullOperationResponse & {
```

Comment:
Discussing with @jeremymeng on the type.

Old _response
![image](https://user-images.githubusercontent.com/10452642/161137909-6c9d522b-4d81-4f4e-a38d-125a4261d595.png)


Compat package doesn't provide HttpResponse.


Proposal
- Add HttpResponse in compat
- Update the object to HttpResponse instead of the FullOperationResponse
- Release compat again
- Use compat in SB and App Config

Reply:
Or we can go with no more changes and just go the `FullOperationResponse` and test the waters with the beta.

Reply:
I'm not opposed to the idea of having the compat package export something like

```ts

export interface CompatRequest {
// fill out
}

export interface CompatHeaders {
// fill out
}

export interface CompatResponse extends FullOperationResponse {
   request: CompatRequest;
   headers: CompatHeaders;
}
```

The packages can then decide to re-export

Reply:
Added `CompatResponse` in `core-http-compat` package and consuming it in app-config.

Can you re-review?

Reply:
@sarangan12 @xirzec @jeremymeng